control-plane/agents/src/bin/ha/cluster/switchover.rs

```diff
@@ -243,48 +247,55 @@ impl SwitchOverEngine {
             nodes,
         };
 
-        sw.mover(rq_rx);
+        sw.mover(Arc::new(Mutex::new(rq_rx)));
         sw
     }
 
-    /// Mover is responsible for processing the switchover request sequentially.
-    pub fn mover(&self, mut recv: UnboundedReceiver<SwitchOverRequest>) {
+    /// Mover instantiates worker task to asynchronously process Switchover request.
+    pub fn mover(&self, recv: Arc<Mutex<UnboundedReceiver<SwitchOverRequest>>>) {
+        for i in 0 .. WORKER_NUM {
+            info!("Starting worker-{} of Switchover Engine", i.to_string());
+            let cloned_self = self.clone();
+            let cloned_channel = Arc::clone(&recv);
+            tokio::spawn(async move { cloned_self.process_request(cloned_channel).await });
+        }
+    }
+
+    /// Switchover request to be handled synchronously in each worker task.
+    async fn process_request(&self, recv: Arc<Mutex<UnboundedReceiver<SwitchOverRequest>>>) {
```

Comment:
think we can just consume self here:
```suggestion
    async fn process_request(self, recv: Arc<Mutex<UnboundedReceiver<SwitchOverRequest>>>) {
```

Reply:
Incorporated